R/validate.R

```diff
@@ -70,12 +78,19 @@ json_validator <- function(schema, engine = "imjv", reference = NULL) {
 ##'   but for now this must be the name of an element within
 ##'   \code{json}.  See the examples for more details.
 ##'
+##' @param strict Set whether the schema should be parsed strictly or not.
+##'   If in strict mode schemas will error to "prevent any unexpected
+##'   behaviours or silently ignored mistakes in user schema". For example
+##'   it will error if encounters unknown formats or unknown keywords. See
+##'   https://ajv.js.org/strict-mode.html for details. Only available in
+##'   \code{engine = 'ajv'}.
+##'
 ##' @export
 ##' @example man-roxygen/example-json_validate.R
 json_validate <- function(json, schema, verbose = FALSE, greedy = FALSE,
-                          error = FALSE, engine = "imjv", reference = NULL,
-                          query = NULL) {
-  tmp <- json_validator(schema, engine, reference = reference)
+                          error = FALSE, engine = "ajv", reference = NULL,
```

Comment:
If we make this change we might break other packages, can you check please?

https://cran.r-project.org/web/packages/jsonvalidate/index.html - see `devtools::revdep` to automate